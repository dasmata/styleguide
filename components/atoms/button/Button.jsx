import React from 'react';
import PropTypes from 'prop-types';
import CX from 'classnames';

import './button.scss';

const prefix = 'a-button';

const Button = ({ text, className, onClick }) => (
  <button
    type="button"
    className={CX([prefix, { [className]: className }])}
    onClick={onClick}
  >
    {text}
  </button>
);

Button.propTypes = {
  text: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func
};

export default Button;
