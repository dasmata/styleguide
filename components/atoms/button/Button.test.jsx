import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';
import simpleButtonConfig from './data/00-simple.json';

describe('Button Atom tests', () => {
  it('Renders a simple button', () => {
    const tree = shallow(<Button {...simpleButtonConfig.moduleConfig} />);
    expect(tree).toMatchSnapshot();
  });
});
