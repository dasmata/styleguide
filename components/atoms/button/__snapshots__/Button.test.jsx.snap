// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Atom tests Renders a simple button 1`] = `
<button
  className="a-button"
  type="button"
>
  Button
</button>
`;
